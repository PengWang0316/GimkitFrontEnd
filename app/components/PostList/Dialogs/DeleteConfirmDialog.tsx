import React, { useCallback } from 'react';
import { connect } from 'react-redux';
import {
  Dialog, DialogTitle, DialogContent, DialogContentText, DialogActions, Button,
} from '@material-ui/core';
import I18n from '@kevinwang0316/i18n';

import { deletePost as deletePostAction } from '../../../store/Posts/actions';

interface Props {
  isOpen: boolean;
  postId: number;
  handleSnackBarCallback: (text: string) => void;
  handleCloseCallback: (event: React.MouseEvent) => void;
  deletePost: (postId: number) => void;
}

export const DeleteConfirmDialog = ({
  isOpen, handleCloseCallback, postId, handleSnackBarCallback, deletePost,
}: Props) => {
  const handleDelete = useCallback(() => {
    deletePost(postId);
    handleCloseCallback(null);
    handleSnackBarCallback(I18n('deletePostSuccess'));
  }, [postId]);
  return (
    <Dialog
      open={isOpen}
      onClose={handleCloseCallback}
    >
      <DialogTitle>{I18n.get('deleteDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText>
          {I18n.get('deleteDialogContent')}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleCloseCallback} color="primary">
          {I18n.get('cancel')}
        </Button>
        <Button onClick={handleDelete} color="primary" autoFocus>
          {I18n.get('delete')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
const mapDispatchToProps = ({
  deletePost: deletePostAction,
});
export default connect(null, mapDispatchToProps)(DeleteConfirmDialog);
