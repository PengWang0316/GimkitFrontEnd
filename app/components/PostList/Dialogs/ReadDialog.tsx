import React, { Fragment, memo } from 'react';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import {
  Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle,
} from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';
import I18n from '@kevinwang0316/i18n';

import { PostType } from '../../../store/Posts/types';

interface Props {
  post: PostType;
  isOpen: boolean;
  handleClose: (event: React.MouseEvent) => void;
}

export const ReadingDialog = ({ post, handleClose, isOpen = false }: Props) => {
  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));

  return (
    <Dialog
      fullScreen={fullScreen}
      open={isOpen}
      onClose={handleClose}
      aria-labelledby="readDialogTitle"
      maxWidth="md"
    >
      {!post && <Fragment>{I18n.get('emptyContent')}</Fragment>}
      {post && (
        <Fragment>
          <DialogTitle id="readDialogTitle">{`${post.title} - ${post.date}`}</DialogTitle>
          <DialogContent>
            <DialogContentText>
              {post.content}
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={handleClose} color="primary">
              {I18n.get('close')}
            </Button>
          </DialogActions>
        </Fragment>
      )}
    </Dialog>
  );
};

export default memo(ReadingDialog);
