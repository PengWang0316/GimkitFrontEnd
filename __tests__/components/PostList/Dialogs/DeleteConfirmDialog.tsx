import React from 'react';
import renderer from 'react-test-renderer';

import { DeleteConfirmDialog } from '../../../../app/components/PostList/Dialogs/DeleteConfirmDialog';

jest.mock('@material-ui/core/Dialog', () => 'Dialog');
jest.mock('@material-ui/core/DialogTitle', () => 'DialogTitle');
jest.mock('@material-ui/core/DialogContent', () => 'DialogContent');
jest.mock('@material-ui/core/DialogContentText', () => 'DialogContentText');
jest.mock('@material-ui/core/DialogActions', () => 'DialogActions');
jest.mock('@material-ui/core/Button', () => 'Button');
jest.mock('@kevinwang0316/i18n', () => ({ get: jest.fn().mockReturnValue('I18n text') }));

describe('DeleteConfirmDialog', () => {
  const defaultProps = {
    isOpen: true,
    handleClose: jest.fn(),
    postId: 1,
    handleSnackBarCallback: jest.fn(),
    deletePost: jest.fn(),
  };
  test('snapshot', () => expect(renderer.create(<DeleteConfirmDialog {...{ ...defaultProps }} />).toJSON()).toMatchSnapshot());
});
