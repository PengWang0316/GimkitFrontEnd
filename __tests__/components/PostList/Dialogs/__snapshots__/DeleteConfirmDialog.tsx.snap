// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirmDialog snapshot 1`] = `
<Dialog
  onClose={[MockFunction]}
  open={true}
>
  <DialogTitle>
    I18n text
  </DialogTitle>
  <DialogContent>
    <DialogContentText>
      I18n text
    </DialogContentText>
  </DialogContent>
  <DialogActions>
    <Button
      color="primary"
      onClick={[MockFunction]}
    >
      I18n text
    </Button>
    <Button
      autoFocus={true}
      color="primary"
      onClick={[Function]}
    >
      I18n text
    </Button>
  </DialogActions>
</Dialog>
`;
